ballerina/json_schema.bal

```diff
@@ -0,0 +1,122 @@
+#
+# + maxLength - The maximum number of characters allowed in the string.
+# + minLength - The minimum number of characters required in the string.
+# + pattern - A regular expression that the string must match.
+# + format - The format that the string must adhere to (e.g., email, date-time).
+public type StringValidationRec record {|
+    int:Unsigned32 maxLength?;
+    int:Unsigned32 minLength?;
+    string:RegExp pattern?;
+    string format?;
+|};
+
+# The annotation is used to specify validation constraints for string type.
+public annotation StringValidationRec StringValidation on type;
+
+# Defines constraints for JSON schema number type.
+#
+# + maximum - The maximum value allowed.
+# + minimum - The minimum value allowed.
+# + exclusiveMaximum - The upper exclusive bound (value must be less than this).
+# + exclusiveMinimum - The lower exclusive bound (value must be greater than this).
+# + multipleOf - The number must be a multiple of this value.
+public type NumberValidationRec record {|
+    int|float|decimal maximum?;
+    int|float|decimal minimum?;
+    int|float|decimal exclusiveMaximum?;
+    int|float|decimal exclusiveMinimum?;
+    int|float|decimal multipleOf?;
+|};
+
+# The annotation is used to specify validation constraints for number type.
+public annotation NumberValidationRec NumberValidation on type;
+
+# Defines constraints for JSON Schema array type.
+#
+# + minItems - The minimum number of items required in the array.
+# + maxItems - The maximum number of items allowed in the array.
+# + uniqueItems - Specifies whether all items in the array must be unique.
+# + unevaluatedItems - Specifies a schema for items that are not evaluated by contains.
+# + contains - Specifies a schema that a specific number of items in the array must match.
+public type ArrayValidationRec record {|
+    int:Unsigned32 minItems = 0;
+    int:Unsigned32 maxItems?;
+    boolean uniqueItems?;
+    record {|
+        int:Unsigned32 minContains = 0;
+        int:Unsigned32 maxContains?;
+        typedesc<json> contains;
+    |} contains?;
+    typedesc<json> unevaluatedItems?;
+|};
+
+# The annotation is used to specify validation constraints for array type.
+public annotation ArrayValidationRec ArrayValidation on type;
+
+public type ObjectValidationRec record {|
+    int:Unsigned32 maxProperties?;
+    int:Unsigned32 minProperties?;
+    typedesc<json> propertyNames?;
+|};
+
+# The annotation is used to specify validation constraints for object type.
+public annotation ObjectValidationRec ObjectValidation on type;
+
+# Defines a dependentSchema.
+#
+# + value - The dependent Sub Schema
+public type DependentSchemaRec record {|
+    typedesc<json> value;
+|};
+
+# The annotation is used to specify a dependent schema.
+public annotation DependentSchemaRec DependentSchema on record field;
+
+# Defines the dependentRequired fields.
+#
+# + value - The dependentRequired fields
+public type DependentRequiredRec record {|
+    string[] value;
```

Comment:
Assuming one value in here is a major usecase.

```
string|string[] value;
```

Reply:
or else don't we need to change this as `values`?

Reply:
+1 for `string|string[] value;`

Reply:
Check other places where arrays are used too.